extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -538,14 +538,21 @@ export async function promptConnectionStringPasswordAndUpdateConnectionString(co
 				connectionString = await vscodeMssqlApi.getConnectionString(connectionDetails, true, false);
 			}
 		}
-		// set connection string to not include the password if connection info does not include password, or user chooses to not include password, or authentication type is not sql login
-		let userPassword: string | undefined;
-		if (includePassword !== constants.yesString) {
+		// set connection string to not include the password if connection info does not include password, 
+		// or user chooses to not include password, or authentication type is not SQL login
+		if (includePassword !== constants.yesString || !connectionInfo.password) {
```

Comment:
Is this going to handle Integrated/AAD auth correctly? (we shouldn't be prompting for password in those scenarios)

Reply:
Yeah, I will add this condition as well to get the getConnectionString with the password set to false. Since we would not need to show the warning or ask user to enter the password manually in that situation.